docs/user_manual.md

```diff
@@ -49,40 +60,341 @@ python -m virtualenv venv
+sources and transformation stages. Each data source has its own set of build
+steps (stages). Datumaro supposes copying of datasets and working in-place by
+default. Modifying operations are recorded in the project, so any of the
+dataset revisions can be reproduced when needed. Multiple dataset versions can
+be stored in different branches with the common data shared.
+
+Let's consider an example of a build tree:
+![build tree](images/build_tree.svg)
+There are 2 data sources in the example project. The resulting dataset
+is obtained by simple merging (joining) the results of the input datasets.
+"Source 1" and "Source 2" are the names of data sources in the project. Each
+source has several stages with their own names. The first stage (called "root")
+represents the original contents of a data source - the data at the
+user-provided URL. The following stages represent operations, which needs to
+be done with the data source to prepare the resulting dataset.
+
+Roughly, such build tree can be created by the following commands (arguments
+are omitted for simplicity):
 ``` bash
-datum --help
+datum create
+
+# describe the first source
+datum add
+datum filter
+datum transform
+datum transform
+
+# describe the second source
+datum add
+datum model add
+datum transform
+datum transform
+
+# now, the resulting dataset can be built with
+datum export
 ```
 
-As a python module:
-> The directory containing Datumaro should be in the `PYTHONPATH`
-> environment variable or `cvat/datumaro/` should be the current directory.
+### Project layout
 
 ``` bash
-python -m datumaro --help
-python datumaro/ --help
-python datum.py --help
+project/
+├── .dvc/
+├── .dvcignore
+├── .git/
+├── .gitignore
+├── .datumaro/
+│   ├── cache/ # object cache
+│   │   ├── <2 leading symbols of obj hash>/
+│   │   │   └── <rest symbols of obj hash>/
+│   │   │       └── <object data>
+│   ├── models/
+│   ├── plugins/ # custom project-specific plugins
+│   │   ├── plugin1/
+│   │   |   ├── __init__.py
+│   │   |   └── file2.py
+│   │   ├── plugin2.py
+│   │   └── ...
+│   ├── tmp/ # temp files
+│   └── tree/ # working tree metadata
+│       ├── config.yml
+│       └── sources/
+│           ├── <source name 1>.dvc
+│           ├── <source name 2>.dvc
+│           └── ...
+│
+├── <source name 1>/
+│   └── <source data>
+└── <source name 2>/
+    └── <source data>
+```
+
+### Use cases
+
+Consider few diagrams describing what Datumaro does for you behind the scene.
+
+Imagine we have a project with 2 data sources. Roughly, it corresponds to
+the following set of commands:
+
+```bash
+datum create
+datum add <...> -n source1
+datum add <...> -n source2
+datum transform <...> source1 # used 3 times
+datum transform <...> source2 # used 5 times
 ```
 
-As a python library:
+Then, for some reasons, the project cache was cleaned from `source1` revisions.
+We also don't have anything in the project working directories.
```

Comment:
Where are the working directories? They should be mentioned in the directory tree in the previous section.

Reply:
Added a descriptive example.